src/components/spreadsheet/utils/equipment-table-editors.js

```diff
@@ -104,6 +105,178 @@ export const GeneratorRegulatingTerminalEditor = forwardRef(
+                            gridContext.dynamicValidation?.ratioTapChanger || {}
+                        );
+                    },
+                    getField: () => {
+                        return colDef.field;
+                    },
+                };
+            },
+            [colDef.field, gridContext.dynamicValidation]
+        );
+
+        const handleSaveRegulatingTerminalPopup = (
+            updatedRegulatedTerminal
+        ) => {
+            const {
+                equipment: { type: equipmentType, id: equipmentId } = {},
+                voltageLevel: { id: voltageLevelId } = {},
+            } = updatedRegulatedTerminal || {};
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                'ratioTapChanger.regulatingTerminalConnectableId',
+                equipmentId
+            );
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                'ratioTapChanger.regulatingTerminalConnectableType',
+                equipmentType
+            );
+            gridContext.dynamicValidation = deepUpdateValue(
+                gridContext.dynamicValidation,
+                'ratioTapChanger.regulatingTerminalVlId',
+                voltageLevelId
+            );
+            setOpenTWTRegulatingTerminalPopup(false);
+        };
+        const handleCancelRegulatingTerminalPopup = () => {
+            setOpenTWTRegulatingTerminalPopup(false);
+            gridApi.stopEditing();
+        };
+
+        const getRatioTapChangerValue = () => {
+            const ratioTapChanger = {
+                ...rowData?.ratioTapChanger,
+                regulatingTerminalConnectableId:
+                    rowData?.ratioTapChanger?.regulatingTerminalConnectableId ||
+                    '',
+                regulatingTerminalConnectableType:
+                    rowData?.ratioTapChanger
+                        ?.regulatingTerminalConnectableType || '',
+                regulatingTerminalVlId:
+                    rowData?.ratioTapChanger?.regulatingTerminalVlId || '',
+            };
+            return ratioTapChanger;
+        };
+
+        return (
+            <RegulatingTerminalModificationDialog
+                open={openTWTRegulatingTerminalPopup}
+                onClose={handleCancelRegulatingTerminalPopup}
+                currentNode={gridContext.currentNode}
+                studyUuid={gridContext.studyUuid}
+                onModifyRegulatingTerminalGenerator={(
+                    updatedRegulatedTerminal
+                ) => {
+                    handleSaveRegulatingTerminalPopup(updatedRegulatedTerminal);
+                }}
+                data={getRatioTapChangerValue()}
+                previousData={gridContext.dataToModify}
+            />
+        );
+    }
+);
+
+export const TWTPhaseRegulatingTerminalEditor = forwardRef(
```

Comment:
Isn't it possible to factorize `TWTPhaseRegulatingTerminalEditor` and `TWTRatioRegulatingTerminalEditor` into something like `TWTRegulatingTerminalEditor`